enhancements/dns/plugins.md

```diff
@@ -97,24 +98,28 @@ resolution, it restricts how cluster operators manage DNS name queries.
 
 ## Proposal
 
-`PluginConfig` defines one or more CoreDNS plugins that can be configured. If a plugin is not defined, it will use a
-default configuration provided by the cluster-dns-operator. A plugin is represented as a slice when multiple instances
-of the plugin can be expressed within the DNS configuration.
+`Server` defines the schema for a server that runs per instance of CoreDNS. A `Server` must have a `Name` that is unique
+among `Servers`. `Zones` defines one or more subdomains to match for name queries to be forwarded to `Upstreams`.
+`ForwardPlugin` provides options for configuring the forward plugin configuration. A `Server` listens on port `5353`,
+which can not be configured at this time. If a `Server` is not specified, the cluster-dns-operator will generate the
+previously depicted ConfigMap.
 
 ```go
-type PluginConfig struct {
-    ForwardPlugins []ForwardPlugin `json:"forwardPlugin"`
+type Server struct {
+    Name string `json:"name"`
+    Zones []string `json:"zones"`
+    ForwardPlugin ForwardPlugin `json:"forwardPlugin"`
```

Comment:
Bikeshed: `Forward` instead of `ForwardPlugin`? The docs could still indicate that it relates to the plugin of the same name.

Reply:
What if `forward`  has a different meaning elsewhere in CoreDNS? The same holds true for other plugin names. My thought was to be explicit with plugin names by adding `Plugin` to the end of the plugin name. Otherwise, I think is makes sense to use the plugin short name if we wrap plugins in a `PluginConfig` struct. Thoughts?

Reply:
Not sure there's a clear winner either way for me yet... do users care if they're configuring a coreDNS plugin, specifically? Would it be just as clear if they were configuring "forwarding"? I'm not sure.

I don't feel strongly, so have no objection to the current proposal.